base/classes.dtx

```diff
@@ -3688,9 +3688,8 @@
 %
 %     A |\caption| command in a `figure' environment writes
 %
-%     |\contentsline{figure}{\numberline{|\meta{num}|}{|%
+%     |\contentsline{figure}{\numberline{|\meta{num}|}{|^^A
 %                    \meta{caption}|}}{|\meta{page}|}|
```

Comment:
taking out the empty line is probably ending up wrong, better surround the whole thing by a quote or raggedleft  environment, then I pull. thanks

Reply:
I think this change is to remove the spurious space just before `<caption>`.
![image](https://user-images.githubusercontent.com/6376638/233176216-29bb74ca-f8d2-4d3b-b3d1-a7f2aaf94036.png)